File: scripts/Sort Colors/Sort Colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        d = {0:0,1:0,2:0}
        for i in nums:
            if i in d:
                d[i]+=1
        
        nums.clear()
        for i in d:
            while d[i]>0:
                nums.append(i)
                d[i]-=1

File: algorithms/Sort Colors/Sort Colors.java
class Solution {

  public void sortColors(int[] nums) {

    int zeroIndex = 0, twoIndex = nums.length - 1, i = 0;
    while (i <= twoIndex) {
      if (nums[i] == 0)
        swap(nums, zeroIndex++, i++);
      else if (nums[i] == 2)
        swap(nums, twoIndex--, i);
      else
        i++;
    }
  }

  public void swap(int[] nums, int i, int j) {
    int temp = nums[i];
    nums[i] = nums[j];
    nums[j] = temp;
  }
}